tiledb/sm/misc/resource_pool.h

```diff
@@ -113,7 +113,10 @@ class ResourcePool {
   resource_handle take() {
     std::lock_guard x(m_);
     if (unused_idx_ == -1)
-      throw std::runtime_error("Ran out of resources in resource pool");
+      throw std::runtime_error(
+          std::string(
+              "Ran out of resources in resource pool with contained type: ") +
+          typeid(T).name());
```

Comment:
Thanks for that! We are working on adding the hierarchical logger in RPs as well, so that we can log in cases of exhaustion to understand where they came from, but I think the info you are "logging" here is complementary to this.
Also FYI we have replaced ResourcePool with BlockingResourcePool in all current usages in core, but I will take care of adding the same type of info about the contained type in BlockingResourcePool exhaustion case when the RP logger is ready.